### PR TITLE
feat(CopyOnClick): add new component

### PR DESCRIPTION
**Pull request by thaytharma**
This PR introduces a new `CopyOnClick`-component that enables text copying. While the functionality for copying already exists as a `Copy` tag-“element” in the Eufemia package, it is not currently exported. The purpose of creating a standalone `CopyOnClick` component is to ensure consistent Eufemia style and behavior across various solutions. We intend to use this for standard texts in our projects.

Please share your thoughts, and I’ll work on advancing this further.

Skjermbilde 2024-08-15 kl  09 15 21

### CODE REVIEW

File: packages/dnb-design-system-portal/src/shared/tags/Copy.module.scss [hunk 1]
```diff
@@ -1,3 +0,0 @@
-.copyStyle {
-  cursor: copy;
-}
```
Comment (tujoworker) [inline packages/dnb-design-system-portal/src/shared/tags/Copy.module.scss:3]: Is this the reason why the cursor not shows up?

**Review by tujoworker — COMMENTED** [review]
Nice work 🫶 We need a discussion if we need a better name of even if it could be a part of another "share" component. Let me know your thoughts/ideas.

Else I will have a chat with the team next week.

File: packages/dnb-design-system-portal/src/shared/tags/Copy.module.scss (hunk 1 above)
Comment (thaytharma) [reply packages/dnb-design-system-portal/src/shared/tags/Copy.module.scss:3]: The style is moved here, so I am not sure why it's not working 🤔
https://github.com/dnbexperience/eufemia/pull/3834/files#diff-6184a503cc2b88e22bdf1a03c8b23131a2eba0fd14612b714036bbef655c401bR6

**Comment by thaytharma** [comment]
> Nice work 🫶 We need a discussion if we need a better name of even if it could be a part of another "share" component. Let me know your thoughts/ideas.
> 
> Else I will have a chat with the team next week.

Please discuss it with the team and let me know.

File: packages/dnb-design-system-portal/src/shared/tags/Copy.module.scss (hunk 1 above)
Comment (thaytharma) [reply packages/dnb-design-system-portal/src/shared/tags/Copy.module.scss:3]: I see that the build file was missing.
Thanks!

File: packages/dnb-design-system-portal/src/docs/uilib/components/copy/info.mdx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+The `Copy` component provides a convenient way to copy text to the clipboard with a single click. This component is particularly useful in scenarios where users need to quickly copy text, such as when copying codes, IDs, URLs, or any other text content that needs to be easily shared or reused.
+Upon hovering, the component can optionally provide visual feedback to the user, displaying a copy cursor or other visual cues that indicate the component's functionality.
+Upon clicking, the component provides a visual feedback to the user, displaying a tooltip with a confirmation message, indicating that the text has been successfully copied.
+
+### Example
+
+Here’s a simple usage example of the `Copy` component:
+
+```tsx
+import { Copy, P } from '@dnb/eufemia'
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/components/copy/info.mdx:16]:
```suggestion
import { Copy, P } from '@dnb/eufemia';
```

File: packages/dnb-design-system-portal/src/docs/uilib/components/copy/info.mdx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+;<P>
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/components/copy/info.mdx:17]:
```suggestion
<P>
```

**Comment by tujoworker** [comment]
Thanks again for making this happen 💯 

We have had a chat internally and have some input:

- Rename it to `CopyOnClick`
- A `copyContent` prop (or similar name) – a dev can use to overwrite the copied value, if it differs from the visually shown value. Can be done in a new PR.
- A `copyIcon` that shows a icon button using the copy icon that is placed at the end of the content, so a user can click to make the copy. Can be done in a new PR.

Also, we need to look into the possibility to move/refactor some code so we share more with `useCopyWithNotice` / `copyWithEffect` used in NumberFormat – that is not in the scope of this PR 😉

**Comment by thaytharma** [comment]
> Thanks again for making this happen 💯
> 
> We have had a chat internally and have some input:
> 
> * Rename it to `CopyOnClick`
> * A `copyContent` prop (or similar name) – a dev can use to overwrite the copied value, if it differs from the visually shown value. Can be done in a new PR.
> * A `copyIcon` that shows a icon button using the copy icon that is placed at the end of the content, so a user can click to make the copy. Can be done in a new PR.
> 
> Also, we need to look into the possibility to move/refactor some code so we share more with `useCopyWithNotice` / `copyWithEffect` used in NumberFormat – that is not in the scope of this PR 😉
> 
> Let me or @langz if this something you want us to take care of 🙏

Thanks for the feedback! 
I’ll be on vacation until next week, so feel free to proceed if you can. 
Otherwise, I’ll look at it when I'm back.

**Comment by thaytharma** [comment]
> > Thanks again for making this happen 💯
> > We have had a chat internally and have some input:
> > 
> > * Rename it to `CopyOnClick`
> > * A `copyContent` prop (or similar name) – a dev can use to overwrite the copied value, if it differs from the visually shown value. Can be done in a new PR.
> > * A `copyIcon` that shows a icon button using the copy icon that is placed at the end of the content, so a user can click to make the copy. Can be done in a new PR.
> > 
> > Also, we need to look into the possibility to move/refactor some code so we share more with `useCopyWithNotice` / `copyWithEffect` used in NumberFormat – that is not in the scope of this PR 😉
> > Let me or @langz if this something you want us to take care of 🙏
> 
> Thanks for the feedback! I’ll be on vacation until next week, so feel free to proceed if you can. Otherwise, I’ll look at it when I'm back.

@tujoworker @langz The component has been renamed to `CopyOnClick`. 
Let's add `copyContent` and `copyIcon` in a different PR. WDYT?

**Comment by langz** [comment]
> > > Thanks again for making this happen 💯
> > > We have had a chat internally and have some input:
> > > 
> > > * Rename it to `CopyOnClick`
> > > * A `copyContent` prop (or similar name) – a dev can use to overwrite the copied value, if it differs from the visually shown value. Can be done in a new PR.
> > > * A `copyIcon` that shows a icon button using the copy icon that is placed at the end of the content, so a user can click to make the copy. Can be done in a new PR.
> > > 
> > > Also, we need to look into the possibility to move/refactor some code so we share more with `useCopyWithNotice` / `copyWithEffect` used in NumberFormat – that is not in the scope of this PR 😉
> > > Let me or @langz if this something you want us to take care of 🙏
> > 
> > 
> > Thanks for the feedback! I’ll be on vacation until next week, so feel free to proceed if you can. Otherwise, I’ll look at it when I'm back.
> 
> @tujoworker @langz The component has been renamed to `CopyOnClick`. Let's add `copyContent` and `copyIcon` in a different PR. WDYT?

Sounds great, I'll take a look at this, and make sure that the build goes green and merge this ASAP :)

**Comment by langz** [comment]
I've:
- Fixed a few imports from Copy to CopyOnClick
- Formatted using prettier
- Added a unit test
- Added screenshot tests for the examples
- Added a own translation for the CopyOnClick component

@tujoworker, please give it a review when you have time 🤞

**Review by tujoworker — APPROVED** [review]
Nice work, Thay. Thank you so much 👏

**Pull request merged**